src/scripts/branch_review.py

```diff
@@ -0,0 +1,312 @@
+        "Children of the obsoletion candidate label",
+        "Children of the obsoletion candidate Definition",
+        "Parent to be obsoleted",
+        "Other direct parents in Branch",
+        "Other direct parents NOT in Branch",
+        "Ancestor inside the branch",
+        "Ancestor outside the branch",
+        "AffectedStatus",
+    ]
+
+    # Open the output file in write mode
+    with open(output_file, "w", newline="") as csvfile:
+        writer = csv.writer(csvfile, delimiter="\t")
+        # Write the column names to the output file
+        writer.writerow(column_names)
+
+        # Convert the obsoletion_candidates list to a set for faster operations
+        obsoletion_candidates_set = set(obsoletion_candidates)
+
+        # Get the relationships of the obsoletion candidates in the current branch
+        obsoletion_candidates_children_relationships = list(
+            OI.relationships(
+                objects=obsoletion_candidates_set, predicates=[IS_A, PART_OF]
+            )
+        )
+
+        # For each branch_id
+        for branch_id in branch_ids:
+            # Get the descendants of the current branch_id
+            branch_descendants_set = branch_descendants_dict[branch_id]
+            obsoletion_candidate_child_of_branch = (
+                branch_descendants_set & obsoletion_candidates_set
+            )
+
+            children_of_obsoletion_candidate = set()
+
+            for obsoletion_candidate in obsoletion_candidate_child_of_branch:
+                # Column E: Parent to be obsoleted == obsoletion_candidate
+                children_of_obsoletion_candidate.update(
+                    set(
+                        get_immediate_children(
+                            curie=obsoletion_candidate,
+                            relationships=obsoletion_candidates_children_relationships,
+                        )
+                    )
+                )
+
+        children_of_obsoletion_candidate_parents_relationship = set(
+            OI.relationships(
+                subjects=children_of_obsoletion_candidate,
+                predicates=[IS_A, PART_OF],
+            )
+        )
+
+        children_also_obsoletion_candidates = (
+            children_of_obsoletion_candidate & obsoletion_candidate_child_of_branch
+        )
+
+        for child in children_of_obsoletion_candidate:
+            # Column B: Children of ObsoleteCandidate == child
+            parents_of_child = (
+                set(
+                    get_immediate_parent(
+                        curie=child,
+                        relationships=children_of_obsoletion_candidate_parents_relationship,
+                    )
+                )
+                - all_obsoletes
+            )
+
+            # These are parents that need to be obsoleted and hence NOT in Branch
+            relevant_obsoletion_parents_in_branch = (
+                parents_of_child & obsoletion_candidate_child_of_branch
```

Comment:
How can this ever be true if `parents_of_child` is `- all_obsoletes`?

Reply:
(I mean, I can see that it works from the table you shared, but I don't get it from the code, sorry :P)

Reply:
All this means is remove all already obsolete terms found by oak.